src/sql/workbench/browser/modelComponents/hyperlink.component.ts

```diff
@@ -21,7 +21,7 @@ import { ILogService } from 'vs/platform/log/common/log';
 
 @Component({
 	selector: 'modelview-hyperlink',
-	template: `<a [href]="url" [title]="title" [attr.aria-label]="ariaLabel" target="blank" [ngStyle]="CSSStyles" [class]="cssClass">{{label}}</a>`
+	template: `<a [href]="url" [title]="title" [attr.aria-label]="ariaLabel" [attr.role]="ariaRole ? ariaRole : null" target="blank" [ngStyle]="CSSStyles" [class]="cssClass">{{label}}</a>`
```

Comment:
Do do `[attr.role]="ariaRole"` - ariaRole is undefined by default already.

Although is there a reason you're doing this instead of making this and the "View All" components actual buttons? You even call them both buttons in the code.

Reply:
This will lead to empty attributes. The view all button looks like this 
`<a target="blank" ng-reflect-ng-style="[object Object]" href="" title="" aria-label="" role="" class="" style="font-size: 13px;">View All</a>`

`<a target="blank" ng-reflect-ng-style="[object Object]" href="" title="" aria-label="" class="" style="font-size: 13px;">View All</a>`

Reply:
The usage of this component is pure stylistic.

Reply:
Empty attributes are fine - they don't harm anything. But I'm curious - does putting null as the value cause it to not add the attribute at all? I've never looked into it but I would have thought that undefined and null would do the same thing (empty attribute) so if that's not the case that's an interesting thing to note...

Reply:
I agree, empty attributes do not harm. I will remove the null check that I am doing. By default, angular should remove the attributes that resolves to undefined or null. However, that does not happen over here.  

Using hyperlinks as button is a style that is derived from the portal. We can surely add an additional style in the button component or use hyperlink component as buttons (like I am doing over here). Just changing the styling won't be enough as we need to color the text blue. 

An example of hyperlink buttons in portal
![image](https://user-images.githubusercontent.com/6816294/126364311-ffefdc26-b0e1-4d52-8ac1-c3ea9c3e1b38.png)

Reply:
What page is that?

In the end I don't really have an issue with using a hyperlink, but it does seem a bit odd (especially since this is one of the only places in the app that I can think of where we do this? Pretty much all other refresh-type actions are actual buttons)

Reply:
https://ms.portal.azure.com/#create/Microsoft.AzureSQL

Reply:
That doesn't even appear to be a11y compliant - not only is the "Show details" not keyboard accessible that I can tell, but it doesn't have any role or aria label associated with it!